query/parse.go

```diff
@@ -210,6 +210,14 @@ func parseExpr(in []byte) (Q, int, error) {
 		// Later we will lift this into a root, like we do for caseQ
 		expr = &Type{Type: t, Child: nil}
 
+	case tokVis:
```

Comment:
nit: this would be slightly more readable for me like this:

```go
switch text {
case "public", "private":
    expr = &Visibility{text}
default:
    return nil, 0, fmt.Errorf("query: unknown visibility argument %q, want {public, private}", text)
}
```